src/core/funcbloc.cpp

```diff
@@ -554,6 +500,21 @@ void CFunctionBlock::nextDataPoint(const CStringDictionary::TStringId *&paDataTy
   }
 }
 
+CAdapter *CFunctionBlock::createAdapter(const SAdapterInstanceDef &paAdapterInstanceDefinition, TForteUInt8 paParentAdapterlistID) {
```

Comment:
Shouldn't this method be now at the CGenFB?

Reply:
Unfortunately, that would create a circular dependency between the definition of `CAdapter` and `CGenFunctionBlock<T>::createAdapter`. Since `CGenFunctionBlock<T>` is a class template, I wouldn't be able to separate the definition from the class (i.e., I cannot put it in a `.cpp` file) like I can in `CFunctionBlock`.

Reply:
oh. Thx for the clarification.